File: jmix-templates/content/component/custom-component/registration.java
package ${packageName};

import io.jmix.flowui.sys.registration.ComponentRegistration;
import io.jmix.flowui.sys.registration.ComponentRegistrationBuilder;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ${registrationClassName} {

    @Bean
    public ComponentRegistration ${xmlElement}() {
        return ComponentRegistrationBuilder.create(${componentClassName}.class)
                .withComponentLoader("${xmlElement}", ${loaderClassName}.class)
                .build();
    }
}

File: jmix-templates/content/component/custom-component/registration.kt
package ${packageName}

import io.jmix.flowui.sys.registration.ComponentRegistration
import io.jmix.flowui.sys.registration.ComponentRegistrationBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class ${registrationClassName} {

    @Bean
    open fun ${xmlElement}(): ComponentRegistration =
        ComponentRegistrationBuilder.create(${componentClassName}::class.java)
            .withComponentLoader("${xmlElement}", ${loaderClassName}::class.java)
            .build()
}
